test/src/test/java/hudson/slaves/DummyCloudImpl.java

```diff
@@ -64,12 +67,29 @@
      */
     public Label label;
 
+    public void setNodeProperties(DescribableList<NodeProperty<?>, NodePropertyDescriptor> nodeProperties) {
+        this.nodeProperties = nodeProperties;
+    }
+    public DescribableList<NodeProperty<?>, NodePropertyDescriptor> getNodeProperties() {
+        return this.nodeProperties;
+    }
+
+    DescribableList<NodeProperty<?>,NodePropertyDescriptor> nodeProperties =
+            new DescribableList<NodeProperty<?>,NodePropertyDescriptor>(Jenkins.getInstance().getNodesObject());
```

Comment:
This is the wrong `owner`.

Why are you bothering with `DescribableList` at all? That is only for UI configuration, which is senseless from a test utility. Plain old `List` would work fine.